Sources/Tracking/DeviceInfo.swift

```diff
@@ -1,15 +1,56 @@
 import Foundation
 #if canImport(UIKit)
 import UIKit
+import UserNotifications
 
-internal enum DeviceInfo {
-    /// Device's model on which SDK is running eg. iPhone12,3
-    static let deviceInfo: String = UIDevice.deviceModelCode
-    /// Operating system and version of OS of the Device
-    static let osInfo: String = "\(UIDevice.current.systemName) \(UIDevice.current.systemVersion)"
-    /// Name of customer's application using the SDK
-    static let customerAppName: String = Bundle.main.infoDictionary?["CFBundleName"] as? String ?? ""
-    /// Version of the customer's application using the SDK
-    static let customerAppVersion: String = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? ""
+/// To get basic detail about the device SDK is working on
+/// such as operating system, customer app name, bundle id etc
+public enum DeviceInfo {
+    case deviceInfo
+    case osInfo
+    case customerAppName
+    case customerAppVersion
+    case customerBundleId
+    case sdkVersion
+    case deviceLocale
+    public var value: String {
+        switch self {
+        case .deviceInfo:
+            return UIDevice.deviceModelCode
+        case .osInfo:
+            return "\(UIDevice.current.systemName) \(UIDevice.current.systemVersion)"
+        case .customerAppName:
+            return Bundle.main.infoDictionary?["CFBundleName"] as? String ?? ""
+        case .customerAppVersion:
+            return Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? ""
+        case .customerBundleId :
+            return Bundle.main.bundleIdentifier ?? ""
+        case .sdkVersion:
+            return SdkVersion.version
+        case .deviceLocale:
+            return Locale.current.identifier
```

Comment:
All of these do not require `UIKit` to collect these. 

Instead of:
```swift
#if canImport(UIKit)
import UIKit

public enum DeviceInfo {
  ... all enum cases in here
}
#endif 
```

we do
```swift
public enum DeviceInfo {
  var value: String? 
    case .deviceInfo:
       #if canimport(uikit)
         return UIDevice.deviceModelCode
        #else 
        return nil 
         #end 
  }
}
```

Another reason why this is great is because you can then remove the `#if canImport(UiKit)` statements in `MessagingPushImplementation.swift`:

```swift
#if canImport(UIKit)
        let deviceOS = DeviceInfo.osInfo.value
        let deviceModel = DeviceInfo.deviceInfo.value
        let appVersion = DeviceInfo.customerAppVersion.value
        let sdkVersion = DeviceInfo.sdkVersion.value
```

You can do:
```swift
if let deviceOs = DeviceInfo.osInfo.value {
  deviceAttributes["deviceOs"] = deviceOs
}
```

Reply:
This has been updated to a new logic !